src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -98,6 +151,24 @@ private void Write(BuildEventArgs e)
                 {
                     eventArgsWriter.Write(e);
                 }
+
+                IncludeSourceFilesFrom(e);
+            }
+        }
+
+        private void IncludeSourceFilesFrom(BuildEventArgs e)
+        {
+            ProjectImportedEventArgs importArgs = e as ProjectImportedEventArgs;
+            if (importArgs != null && importArgs.ImportedProjectFile != null)
+            {
+                projectImportsCollector.AddFile(importArgs.ImportedProjectFile);
```

Comment:
If I set `CollectProjectImports` to `ProjectImportsCollectionMode.None` will this be a null ref?